### ISSUE TITLE
Log methods should not use print

### ISSUE THREAD

**Issue by mrkevinze** [stripe/stripe-python#274]
When verbose logging is enabled, it was surprising that the details are printed to the console via `print` and are not part of the log message itself.

This is the current code in `util.py`:

```
def log_debug(message, **params):
    if _console_log_level() == 'debug':
        print(logfmt(dict(message=message, **params)))
    logger.debug(message, params)


def log_info(message, **params):
    if _console_log_level() in ['debug', 'info']:
        print(logfmt(dict(message=message, **params)))
    logger.info(message, params)
```

This causes problems when the application needs to filter through its logs, because the individual print statements get elevated to severity level error (at least on App Engine). Also, passing `params` to the log method does not change the output log statement, i.e. only the message will be logged and not its details.

My suggested fix is to replace the print statement with a log statement as follows, thereby making the entire method rely on the logging facility:

```
def log_debug(message, **params):
    if _console_log_level() == 'debug':
        logger.debug(message + "\n" + logfmt(dict(**params)))
    else:
        logger.debug(message)


def log_info(message, **params):
    if _console_log_level() in ('debug', 'info'):
        logger.info(message + "\n" + logfmt(dict(**params)))
    else:
        logger.info(message)
```

Python 2.7.13
Stripe 1.44.0

**Comment by brandur-stripe** [comment]
Thanks for the report!

cc @deontologician-stripe Can you take a look at this please?

**Comment by deontologician-stripe** [comment]
It seems like there are two behaviors under discussion here. The first is that `print` and `logger.debug` output different information. I agree this is definitely wrong and I'll fix it.

The second is the behavior that if you use `STRIPE_LOG` or `stripe.log`, you'll always get logs to the console (it uses `print` under the hood). This is intentional, the main reason for it is to always have an easy way to see what is being sent to and from the Stripe API without having to configure logging. If you're in a situation where anything printed to the console is considered an error, you should definitely set up logging explicitly (using `logging.basicConfig()` and a custom handler) and not use the `STRIPE_LOG` or `stripe.log` settings.

Although on second thought, it might make sense to ensure the console logging does go to `stderr` and not `stdout`. That is probably what App Engine is triggering on

**Comment by deontologician-stripe** [comment]
OK, proposed fix up in #275

**Comment by mrkevinze** [comment]
Thanks for the followup! I see the intention behind using print now. For production code, I will probably set Stripe's logger explicitly rather than use `STRIPE_LOG` or `stripe.log`, and set the level to `logging.INFO` to keep customer data out of the logs.

**Comment by deontologician-stripe** [comment]
Ok, fix is merged, and deployed in version [1.45.0](https://pypi.python.org/pypi/stripe/1.45.0)

**Comment by deontologician-stripe** [comment]
Also, thanks @mrkevinze for bringing opening the issue, it's much more polished now :)
